api/v1alpha1/healthcheck_types.go

```diff
@@ -152,6 +152,7 @@ type HTTPActiveHealthChecker struct {
 	// Hostname defines the HTTP host that will be requested during health checking.
 	// Default: HTTPRoute or GRPCRoute hostname.
 	//
+	// +kubebuilder:validation:Optional
```

Comment:
you need to run `make kube-generate` to reflect these changes in the CRD (the `git status` would show a modified generated file)